src/controller/c/robot.c

```diff
@@ -447,10 +447,9 @@ void robot_read_answer(WbDevice *d, WbRequest *r) {
       break;
     case C_ROBOT_WWI_MESSAGE:
       n = request_read_int32(r);
-      if (n > robot.wwi_message_received_size)
-        robot.wwi_message_received = realloc(robot.wwi_message_received, n);
-      robot.wwi_message_received_size = n;
-      memcpy(robot.wwi_message_received, request_read_data(r, n), n);
+      robot.wwi_message_received = realloc(robot.wwi_message_received, robot.wwi_message_received_size + n);
```

Comment:
Calling realloc systematically is costly in terms of performance. We should realloc only if the buffer size is too small, not if it's too big.